testing/test_session.py

```diff
@@ -274,16 +274,26 @@ def test_deselect(testdir):
     testdir.makepyfile(
         test_a="""
         import pytest
+
         def test_a1(): pass
+
         @pytest.mark.parametrize('b', range(3))
         def test_a2(b): pass
+
+        class TestClass:
+            def test_c1(self): pass
+
+            def test_c2(self): pass
     """
     )
     result = testdir.runpytest(
-        "-v", "--deselect=test_a.py::test_a2[1]", "--deselect=test_a.py::test_a2[2]"
+        "-v",
+        "--deselect=test_a.py::test_a2[1]",
+        "--deselect=test_a.py::test_a2[2]",
+        "--deselect=test_a.py::TestClass::test_c1",
```

Comment:
I wonder if we should still support the old syntax, but figured it would not be worth it really.
People might be affected though if they use this in scripts already etc..?!
Anyway, it currently uses `.startswith()`, and therefore a `Node.matches` method should be added/used.

Reply:
I agree it is not worth supporting the old syntax. 👍